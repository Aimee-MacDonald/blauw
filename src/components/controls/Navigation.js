import React from 'react'
import styled from 'styled-components'

const Navigation = props => (
  <StyledNavigation {...props}
    onMouseEnter={props.toggleOpen}
    onMouseLeave={props.toggleOpen}
  >
    {props.open && <h1>Navigation</h1>}
  </StyledNavigation>
)

const StyledNavigation = styled.nav`
  background-color: ${props => props.open ? 'pink' : 'transparent'};
  width: ${props => props.open ? 16 : 0.4}rem;
  height: 96vh;
  overflow-x: hidden;
  transition: 0.4s;
  border-top-right-radius: 0.4rem;
  border-bottom-right-radius: 0.4rem;
  padding: 1rem;
  box-shadow: ${props => props.open ? '0.2rem 0 1rem 0 #248' : 0};
`

export default Navigation