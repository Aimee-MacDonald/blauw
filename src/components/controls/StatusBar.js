import React from 'react'
import styled from 'styled-components'
import {connect} from 'react-redux'

const StatusBar = props => (
  <StyledStatusBar {...props}>
    {!props.connected && <p>Unable to Connect to the Local Server</p>}
  </StyledStatusBar>
)

const StyledStatusBar = styled.div`
  position: fixed;
  width: 100%;
  height: ${props => !props.connected ? '1rem' : '0'};
  background-color: ${props => props.connected ? 'lime' : 'orangered'};
  transition: 0.4s;

  p{
    text-align: center;
  }
`

const mapStateToProps = ({util}) => ({...util})

export default connect(mapStateToProps)(StatusBar)