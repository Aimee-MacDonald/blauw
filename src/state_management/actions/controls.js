export const toggleControl = orientation => ({
  type: 'TOGGLE_CONTROL',
  payload: orientation
})