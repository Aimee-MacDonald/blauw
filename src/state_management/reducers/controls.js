const defaultState = {
  left: {
    open: false
  },
  right: {
    open: false
  }
}

export default (state = defaultState, {type, orientation}) => {
  switch(type){
    case 'TOGGLE_CONTROL':
      if(orientation === 'left'){
        return{
          ...state,
          left: {
            open: !state.left.open
          }
        }
      } else if(orientation === 'right'){
        return{
          ...state,
          right: {
            open: !state.right.open
          }
        }
      }

    default:
      return state
  }
}