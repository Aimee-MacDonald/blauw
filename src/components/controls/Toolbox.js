import React from 'react'
import styled from 'styled-components'
import {connect} from 'react-redux'

import {toggleControl, toggleControlLock} from '../../state_management/actions/controls'
import {setModal} from '../../state_management/actions/modal'
import {getMainsFlags} from '../../state_management/selectors/navigation'
import {isToday} from '../../util/util'
import {getSelectedBooking} from '../../state_management/selectors/bookings'

export const Toolbox = props => (
  <StyledToolbox {...props}
    onMouseEnter={() => props.dispatch(toggleControl('right'))}
    onMouseLeave={() => props.dispatch(toggleControl('right'))}
  >
    
    {props.open && <button onClick={() => props.dispatch(toggleControlLock('right'))}>Lock</button>}

    {props.open && (
      <div>
        {props.navigation.bookingSheet && (
          <div>
            <button onClick={() => props.dispatch(setModal('createBooking'))}>Create Booking</button>
            {props.selectedBooking && (
              <div>
                <button onClick={() => props.dispatch(setModal('deleteBooking'))}>Delete Booking</button>
                <button onClick={() => props.dispatch(setModal('editBooking'))}>Edit Booking</button>
                {isToday(props.selectedBooking.checkin_date) && !props.selectedBooking.checked_in && props.selectedBooking.status === 'confirmed' && <button onClick={() => props.dispatch(setModal('checkin'))}>Check In</button>}
                {props.selectedBooking.checked_in && <button onClick={() => props.dispatch(setModal('checkout'))}>Check Out</button>}
                {props.selectedBooking.status === 'checked_out' && <button onClick={() => props.dispatch(setModal('undoCheckout'))}>Undo Check Out</button>}
              </div>
            )}
          </div>
        )}

        {props.navigation.editRoomsList &&
          <div>
            <button onClick={() => props.dispatch(setModal('createRoom'))}>Add Room</button>
            <button onClick={() => props.dispatch(setModal('removeRoom'))}>Remove Room</button>
            <button onClick={() => props.dispatch(setModal('editRoom'))}>Edit Room</button>
          </div>
        }

        {props.navigation.notes &&
        <div>
          <button onClick={() => props.dispatch(setModal('createNote'))}>Add Note</button>
        </div>
        }
      </div>
    )}
  </StyledToolbox>
)

const StyledToolbox = styled.nav`
  background-color: ${props => props.open ? 'pink' : 'transparent'};
  width: ${props => props.open ? 16 : 0.4}rem;
  height: 96vh;
  overflow-x: hidden;
  transition: 0.4s;
  border-top-left-radius: 0.4rem;
  border-bottom-left-radius: 0.4rem;
  padding: 1rem;
  box-shadow: ${props => props.open ? '-0.2rem 0 1rem 0 #248' : 0};
`

const mapStateToProps = ({controls, navigation, bookings}) => {
  return{
    open: controls.right.open,
    navigation : getMainsFlags(navigation),
    selectedBooking: getSelectedBooking(bookings)
  }
}

export default connect(mapStateToProps)(Toolbox)