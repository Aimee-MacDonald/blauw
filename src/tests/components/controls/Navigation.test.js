import React from 'react'
import {shallow} from 'enzyme'

import {Navigation} from '../../../components/controls/Navigation'

test('Render closed Navigation', () => {
  const wrapper = shallow(<Navigation open={false}/>)
  expect(wrapper.getElement()).toMatchSnapshot()
})

test('Render opened Navigation', () => {
  const wrapper = shallow(<Navigation open={true}/>)
  expect(wrapper.getElement()).toMatchSnapshot()
})