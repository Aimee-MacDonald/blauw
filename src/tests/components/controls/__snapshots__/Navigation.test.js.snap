// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render closed Navigation 1`] = `
<styled.nav
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  open={false}
/>
`;

exports[`Render opened Navigation 1`] = `
<styled.nav
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  open={true}
>
  <button
    onClick={[Function]}
  >
    <BsLockFill />
  </button>
  <div>
    <button
      onClick={[Function]}
    >
      Booking Sheet
    </button>
    <button
      onClick={[Function]}
    >
      Checkout
    </button>
    <button
      onClick={[Function]}
    >
      Stock
    </button>
    <button
      onClick={[Function]}
    >
      Notes
    </button>
    <button
      onClick={[Function]}
    >
      Edit Rooms List
    </button>
  </div>
</styled.nav>
`;
